biz.aQute.bndlib.tests/test/test/component/DSAnnotationTest.java

```diff
@@ -4444,6 +4443,25 @@ public void anyservice() throws Exception {
 		}
 	}
 
+	@Test
+	public void anyserviceNoServiceRequirement() throws Exception {
+		try (Builder b = new Builder()) {
+			b.setProperty(Constants.DSANNOTATIONS, "test.component.DSAnnotationTest$AnyServiceUse");
+			b.setProperty("Private-Package", "test.component");
+			b.addClasspath(new File("bin_test"));
+
+			Jar jar = b.build();
+			assertOk(b);
+			Attributes a = getAttr(jar);
+			checkProvides(a);
+
+			jar.getManifest()
+				.write(System.out);
+
+			assertThat(a.getValue(Constants.REQUIRE_CAPABILITY)).doesNotContain("osgi.service");
```

Comment:
This check it too broad. It expects no osgi.service requirement. Changes to the test component in the future could break this. Perhaps just ensuring the require capability header does not contain 'AnyService.class.getName()' is better.

```suggestion
			assertThat(a.getValue(Constants.REQUIRE_CAPABILITY)).doesNotContain(AnyService.class.getName());
```

Reply:
Ahh, you are already on it. Thanks heaps.